src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -341,6 +341,128 @@ public BitArray Not()
             return this;
         }
 
+        /*=========================================================================
+        ** Shift all the bit values to right on count bits. The current instance is
+        ** updated and returned.
+        * 
+        ** Exceptions: ArgumentOutOfRangeException if count < 0
+        =========================================================================*/
+        public BitArray RightShift(int count)
+        {
+            if (count <= 0)
+            {
+                if (count < 0)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(count), count, SR.ArgumentOutOfRange_NeedNonNegNum);
+                }
+                Contract.EndContractBlock();
+
+                _version++;
+                return this;
+            }
```

Comment:
```cs
if (count <= 0)  { 
     if (count < 0) { } 
}
```
Made 1 comparison in usual flow and 2 in rare flow.

```cs
if (count == 0) { }
if (count < 0) { }
```
Made 2 comparison in usual flow.